### DISCUSSION TITLE
CodePipeline – Include git submodule from Bitbucket source

### DISCUSSION THREAD

**Discussion by g-danna** [aws/aws-cdk#24470] [Q&A]
I have a CodePipeline that uses a Bitbucket repo as its main source.

```javascript
const pipeline = new pipelines.CodePipeline(this, id, {
    crossAccountKeys: false,
    synth: new pipelines.CodeBuildStep("Synth", {
        input: pipelines.CodePipelineSource.connection(
            "<repo name>",
            "<branch name>"
            // ConnectionSourceOptions
            {
                connectionArn: "<connection arn>",
                // codeBuildCloneOutput – if true, it adds whole git repo
            }
        ),
        buildEnvironment: {
            environmentVariables: { ... },
        },
        commands: [
            "cd src",
            "npm install",
            "npm run build",
            "cd ../stack",
            "npm install",
            "npx cdk synth",
        ],
        primaryOutputDirectory: "stack/cdk.out",
    }),
});
```

There is also a "Connection" (under "Developer Tools") – a trust – between AWS and Bitbucket in order to avoid using SSH keys.

I enabled git on the source using `codeBuildCloneOutput: true` in the [`ConnectionSourceOptions`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_pipelines.ConnectionSourceOptions.html). However, when I run commands like `git submodule init` and `git submodule update --remote` the trust fails. It throws an 'Unauthorized' fatal error.

I don't want to use SSH keys – company policy is to keep things simple and rely on the Connection between AWS and Bitbucket. Is there a way to make git submodules work within this trust, without SSH keys?

**Comment by devsheva** [comment]
i'm encountering the same concern. did you find a workaround?
